### PR TITLE
Implement WhitelistService and improved BanService

### PR DESCRIPTION
**Pull request by Aaron1011**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/954) | SpongeCommon

Implementation of `WhitelistService` and improved `BanService`.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/ban/SpongeBanBuilder.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    private BanType banType;
+    private Text reason;
+    private Date start = new Date();
+    private Date end;
+    private Text source;
+
+    @Override
+    public Ban.Builder profile(org.spongepowered.api.profile.GameProfile profile) {
+        checkNotNull(profile, "Profile cannot be null!");
+        checkState(banType == BanTypes.PROFILE, "Cannot set a GameProfile if the BanType is not BanType.USER_BAN!");
```
Comment (Deamon5550) [inline src/main/java/org/spongepowered/common/ban/SpongeBanBuilder.java]: Message needs updating

File: src/main/java/org/spongepowered/common/ban/SpongeBanBuilder.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        this.profile = profile;
+        return this;
+    }
+
+    @Override
+    public Ban.Builder address(InetAddress address) {
+        checkNotNull(address, "Address cannot be null!");
+        checkState(banType == BanTypes.IP, "Cannot set an InetAddress if the BanType is not BanType.IP_BAN!");
```
Comment (Deamon5550) [inline src/main/java/org/spongepowered/common/ban/SpongeBanBuilder.java]: Same here

File: src/main/java/org/spongepowered/common/SpongeBootstrap.java [hunk 3]
```diff
@@ -54,7 +58,8 @@
 import java.util.UUID;
 
 /**
- * Used to setup the ecosystem.
+ * Used to setup the ecosyst
+ * em.
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/SpongeBootstrap.java]: "Oops."

**Comment by Aaron1011** [comment]
Closed in https://github.com/SpongePowered/SpongeCommon/commit/2465d1a260f90f7b166133b08ff5ecfd3b6aa17a

**Pull request closed without merging**